src/main/java/io/jenkins/plugins/pipelinegraphview/utils/PipelineNodeUtil.java

```diff
@@ -47,11 +50,43 @@ public static String getDisplayName(@NonNull FlowNode node) {
                 : name;
     }
 
+    public static boolean isStep(FlowNode node) {
+        if (node != null) {
+            if (node instanceof AtomNode) {
+                return true;
+            }
+            if (node instanceof StepStartNode) {
+                StepStartNode stepStartNode = (StepStartNode) node;
+                boolean takesImplicitBlockArgument = false;
+                StepDescriptor sd = stepStartNode.getDescriptor();
+                if (sd != null) {
+                    takesImplicitBlockArgument = sd.takesImplicitBlockArgument();
+                }
+                return !isStage(node)
+                        && !isParallelBranch(node)
+                        && stepStartNode.isBody()
+                        && !takesImplicitBlockArgument;
+            }
+        }
+        return false;
+    }
+
     public static boolean isStage(FlowNode node) {
-        return node != null
-                && ((node.getAction(StageAction.class) != null)
-                        || (node.getAction(LabelAction.class) != null
-                                && node.getAction(ThreadNameAction.class) == null));
+        if (node != null) {
+            if (node instanceof StepStartNode) {
+                StepStartNode stepStartNode = (StepStartNode) node;
+                if (stepStartNode.getDescriptor() != null) {
+                    StepDescriptor sd = stepStartNode.getDescriptor();
+                    return sd != null && StageStep.DescriptorImpl.class.equals(sd.getClass()) && stepStartNode.isBody();
+                }
+            }
+            LabelAction labelAction = node.getAction(LabelAction.class);
+            ThreadNameAction threadNameAction = node.getAction(ThreadNameAction.class);
+            return labelAction != null
+                    && PARALLEL_SYNTHETIC_STAGE_NAME.equals(labelAction.getDisplayName())
+                    && threadNameAction == null;
```

Comment:
Keep the old logic only to include the "synthetic parallel" stage that comes from calling `parallel` outside of a `stage` block.